src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -0,0 +1,266 @@
+
+            // test if we are at a function call
+            if (!$token->isGivenKind(T_STRING)) {
+                continue;
+            }
+
+            $next = $tokens->getNextMeaningfulToken($index);
+            if (!$tokens[$next]->equals('(')) {
+                continue;
+            }
+
+            $functionNamePrefix = $tokens->getPrevMeaningfulToken($index);
+            if ($tokens[$functionNamePrefix]->isGivenKind(array(T_DOUBLE_COLON, T_NEW, T_OBJECT_OPERATOR, T_FUNCTION))) {
+                continue;
+            }
+
+            if ($tokens[$functionNamePrefix]->isGivenKind(T_NS_SEPARATOR)) {
+                // skip if the call is to a constructor or to a function in a namespace other than the default
+                $prev = $tokens->getPrevMeaningfulToken($functionNamePrefix);
+                if ($tokens[$prev]->isGivenKind(array(T_STRING, T_NEW))) {
+                    continue;
+                }
+            }
+
+            $lowerFunctionName = \strtolower($tokenContent);
+
+            if (!\in_array($lowerFunctionName, $internalFunctionNames, true) || \in_array($lowerFunctionName, $this->exclude, true)) {
+                continue;
+            }
+
+            // do not bother if previous token is already namespace separator
+            if ($tokens[$index - 1]->isGivenKind(T_NS_SEPARATOR)) {
+                continue;
+            }
+
+            $indexes[] = $index;
+        }
+
+        $indexes = \array_reverse($indexes);
+        foreach ($indexes as $index) {
+            $tokens->insertAt($index, new Token(array(T_NS_SEPARATOR, '\\')));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Add leading `\` before function invocation of internal function within namespaces to speed up resolving.',
+            array(
+                new CodeSample(
+'<?php
+
+namespace Foo;
+
+class Bar
+{
+    public function baz($options) 
+    {
+        if (!array_key_exists("foo", $options)) {
+            throw new \InvalidArgumentException();
+        }
+        
+        return json_encode($options);
+    }
+    
+}'
+                ),
+                new CodeSample(
+'<?php
+
+namespace Foo;
+
+class Bar
+{
+    public function baz($options) 
+    {
+        if (!array_key_exists("foo", $options)) {
+            throw new \InvalidArgumentException();
+        }
+        
+        return json_encode($options);
+    }
+    
+}',
+                    array(
+                        'exclude' => array(
+                            'json_encode',
+                        ),
+                    )
+                ),
+            ),
+            null,
+            'Configure names of functions to exclude, for example, when using a polyfill.',
+            self::$defaultConfiguration,
+            'Risky if a function with the same name as a native function exists in the current namespace, or a polyfill is used.'
```

Comment:
maybe sth like this ?

```diff
-'Risky if a function with the same name as a native function exists in the current namespace, or a polyfill is used.'
+'Rule is risky when a function with the same name as a native function exists in the current namespace.
+Two major situations when it could happen are:
+* polyfill is used to provide function from higher PHP version, if PHP CS Fixer would been run also on that higher PHP version, but software is also desired to work on lower PHP version that naively doesn't have that polyfilled function
+* function is mocked during tests execution, eg mocking `time` function - in that case after applying the rule src code will always use original, unmocked function
+To deal with described situation provide a configuration with function names you want to preserve unchanged.'
```